### PR TITLE
appsody stack: generate json index

### PR DESCRIPTION
**Pull request by skoh7645**
Generates an `index.json` file for CodeWind in the commands:
- `stack package`
- `stack add-to-repo`
- `stack remove-from-repo`

Fixes: https://github.com/appsody/appsody/issues/908

### CODE REVIEW

File: cmd/utils.go [hunk 1]
```diff
@@ -2322,3 +2322,39 @@ func RemoveIfExists(path string) error {
...
+			indexJSONStack = append(indexJSONStack, stackJSON)
+		}
+	}
+
+	// Last thing to do is write the data to the file
+	data, err := json.MarshalIndent(&indexJSONStack, "", "	")
+	if err != nil {
+		return err
+	}
+	indexFilePath = strings.ReplaceAll(indexFilePath, ".yaml", ".json")
```
Comment (EnriqueL8) [inline cmd/utils.go]: We could be stricter here by only replacing once

File: cmd/utils.go [hunk 2]
```diff
@@ -2322,3 +2322,39 @@ func RemoveIfExists(path string) error {
 	}
 	return nil
 }
+
+func generateJSON(log *LoggingConfig, indexYaml IndexYaml, indexFilePath string) error {
```
Comment (EnriqueL8) [inline cmd/utils.go]: We could rename this function to generateCodewindJSON, because the format is specific to them.

File: cmd/utils.go [hunk 3]
```diff
@@ -2322,3 +2322,39 @@ func RemoveIfExists(path string) error {
...
+			stackJSON := IndexJSONStack{}
+			stackJSON.DisplayName = "Appsody " + stack.Name + " " + template.ID + " template"
+			stackJSON.Description = stack.Description
+			stackJSON.Language = stack.Language
+			stackJSON.ProjectType = "appsodyExtension"
+			stackJSON.ProjectStyle = "Appsody"
+			stackJSON.Location = template.URL
+
+			link := Link{}
+			link.Self = "/devfiles/" + stack.ID + "/devfile.yaml"
```
Comment (EnriqueL8) [inline cmd/utils.go]: As @CameronMcWilliam mentioned, could we use `filepath.Join`?

File: cmd/stack_package.go [hunk 4]
```diff
@@ -76,6 +76,21 @@ type IndexYamlStackTemplate struct {
...
+	DisplayName  string `json:"displayName"`
+	Description  string `json:"description"`
+	Language     string `json:"language"`
+	ProjectType  string `json:"projectType"`
+	ProjectStyle string `json:"projectStyle"`
+	Location     string `json:"location"`
+	Links        Link   `json:"links"`
+}
+
+type Link struct {
```
Comment (EnriqueL8) [inline cmd/stack_package.go]: we could just rename this to `Links`

File: cmd/utils.go (hunk 3 above)
Comment (skoh7645) [reply cmd/utils.go]: Using `filepath.Join` on Windows would result in backslashes, which we don't want. See [here](https://github.com/kabanero-io/codewind-templates/blob/master/devfiles/index.json) for a valid Codewind index

**Comment by upLukeWeston** [comment]
The documentation will need to be updated to reflect this new feature.

**Pull request merged**